libfoundation/src/foundation-string.cpp

```diff
@@ -4983,9 +4987,60 @@ bool MCStringConvertToSysString(MCStringRef p_string, const char * &r_system_str
     t_sys_string[t_sys_len] = '\0';
 
 	r_system_string = t_sys_string;
+    r_byte_count = t_sys_len;
+    
 	return true;
 }
 
+#elif defined(__MAC__)
+bool MCStringConvertToSysString(MCStringRef p_string, char *& r_system_string, size_t& r_byte_count)
+{
+    bool t_success;
+    uindex_t t_byte_count;
+    if (!MCStringConvertToUTF8(p_string, r_system_string, t_byte_count))
+        return false;
```

Comment:
If you look at the implementation of MCStringConvertToUTF8 - I'm pretty sure the return parameter is the byte-count and not char count... (the parameter name in the header is misleading).